assets/src/edit-story/components/panels/slug.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Row, TextInput, HelperText } from '../form';
+import { useStory } from '../../app/story';
+import { SimplePanel } from './panel';
+
+const Permalink = styled.a`
+  color: ${({ theme }) => theme.colors.link};
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px 6px;
+  border-radius: 4px;
+  flex-grow: 1;
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.fg.v1};
+  }
+`;
+
+function SlugPanel() {
+  const {
+    state: {
+      story: { slug, link },
+    },
+    actions: { updateStory },
+  } = useStory();
+  const handleChangeValue = useCallback(
+    (prop) => (value) => updateStory({ properties: { [prop]: value } }),
+    [updateStory]
+  );
+  return (
+    <SimplePanel name="permalink" title={__('Permalink', 'web-stories')}>
+      <Row>
+        <BoxedTextInput
+          label={__('URL Slug', 'web-stories')}
+          value={slug}
+          onChange={handleChangeValue('slug')}
+          placeholder={__('Enter slug', 'web-stories')}
+        />
+      </Row>
+      <HelperText>
+        <Permalink rel="noopener noreferrer" target="_blank" href={link}>
+          {link}
+        </Permalink>
```

Comment:
So this doesn't actually show the slug, that I've input. But that's due to some other WP setting, or?
![Screen Shot 2020-03-23 at 3 39 40 PM](https://user-images.githubusercontent.com/637548/77356423-a4d29f00-6d1c-11ea-8f58-f520c494a273.png)

Reply:
Yeah, shouldn't this live update?

Reply:
The link of the post doesn't display "pretty" if the post isn't published. However, currently, it doesn't live update the link anyway, only upon saving, since the link might change after saving (if there are duplicates). So if you have the post published and pretty permalink turned on then it should display the correct link after saving.

(Perhaps it should live update the estimated link assuming there are no duplicates though.)

Reply:
Live update added: 1bbd3f9